### ISSUE TITLE
No actual inputs buffer

### ISSUE THREAD

**Issue by kemolumi** [helloimGlace/snek#3]
The input logic is called between renders, which would only check for input when a new frame is drawn, causing snek to *sometimes* follow player's input, make things not snappy -> makes snek sad.

Try using `Gdx.input.setInputProcessor()`, it will run based on the input event itself.

Then, pass the initialized input processor instance to the `GameScreen`, and read the buffered value created in the class, removing the need of `tempDirection` and `move` variables, plus the whole `input()` method.

<details>
  <summary>
    <b>InputBufferer</b> class
  </summary>

```java
package io.github.snek;

import com.badlogic.gdx.InputProcessor;
import com.badlogic.gdx.utils.Array;

import java.util.HashMap;

public class InputBufferer implements InputProcessor {
    public enum Direction {
        UP,
        DOWN,
        RIGHT,
        LEFT,
    }

    private HashMap<Integer, Direction> directionMap = new HashMap<Integer, Direction>() {{
        put(19, Direction.UP);
        put(20, Direction.DOWN);
        put(21, Direction.LEFT);
        put(22, Direction.RIGHT);
    }};
    private HashMap<Direction, Direction> blacklisted = new HashMap<Direction, Direction>() {{
        put(Direction.UP, Direction.DOWN);
        put(Direction.DOWN, Direction.UP);
        put(Direction.LEFT, Direction.RIGHT);
        put(Direction.RIGHT, Direction.LEFT);
    }};

    public Array<Direction> inputBuffer = new Array<>();
    private Direction currentDirectionInput = Direction.RIGHT;

    // Only process the key down event, returns true to absorb the input.
    @Override
    public boolean keyDown(int keycode) {
        Direction inputDirection = directionMap.get(keycode);
        if (
            inputDirection != null && 
            inputDirection != currentDirectionInput && 
            blacklisted.get(currentDirectionInput) != inputDirection
        ) {
            currentDirectionInput = inputDirection;
            inputBuffer.add(inputDirection);
        }

        return true;
    }

    @Override
    public boolean keyUp(int keycode) {
        return false;
    }

    @Override
    public boolean keyTyped(char character) {
        return false;
    }

    @Override
    public boolean touchDown(int screenX, int screenY, int pointer, int button) {
        return false;
    }

    @Override
    public boolean touchUp(int screenX, int screenY, int pointer, int button) {
        return false;
    }

    @Override
    public boolean touchCancelled(int screenX, int screenY, int pointer, int button) {
        return false;
    }

    @Override
    public boolean touchDragged(int screenX, int screenY, int pointer) {
        return false;
    }

    @Override
    public boolean mouseMoved(int screenX, int screenY) {
        return false;
    }

    @Override
    public boolean scrolled(float amountX, float amountY) {
        return false;
    }
}
```

</details>

<details>
  <summary>
    Initialize
  </summary>

```java
InputBufferer inputBufferer = new InputBufferer();
// Remove previously initialized processor.
Gdx.input.setInputProcessor(null);
Gdx.input.setInputProcessor(inputBufferer);
game.setScreen(new GameScreen(game, inputBufferer));
```

</details>

<details>
  <summary>
    Edit: <b>switch case</b> does work with enums, so no need to pretend one by declaring chars.
  </summary>

```java
// Direction "enums" | (Unused)
// private static final char RIGHT = 'r';
// private static final char LEFT = 'l';
// private static final char UP = 'u';
// private static final char DOWN = 'd';

// `direction` will use `InputBufferer.Direction` instead.
switch (direction) {
    case RIGHT: {
        snekX += grid;
        return;
    }
    case LEFT: {
        snekX -= grid;
        return;
    }
    case UP: {
        snekY += grid;
        return;
    }
    case DOWN: {
        snekY -= grid;
    }
}
```

</details>

snek 👍

**Comment by helloimGlace** [comment]
Actually I was working on a much, much worse implementation of an input queuing system. Thanks for the feedback and all the suggestions. I'll try to have them in the main branch.

Also I completely forgot that you do follow me on GitHub. F**k you.

**Comment by helloimGlace** [comment]
Plus, the enum thing was just me being dumb, i got it fixed later on lol.

**Comment by helloimGlace** [comment]
And if you can, please give more of your suggestions and create a pull request into a testing branch.

**Comment by kemolumi** [comment]
Fixed in #5 

Merged at 2fd5be4
